indexing-hadoop/src/main/java/io/druid/indexer/DetermineHashedPartitionsJob.java

```diff
@@ -324,12 +324,9 @@ protected void reduce(
             HyperLogLogCollector.makeCollector(ByteBuffer.wrap(value.getBytes(), 0, value.getLength()))
         );
       }
-      Optional<Interval> intervalOptional = config.getGranularitySpec().bucketInterval(new DateTime(key.get()));
 
-      if (!intervalOptional.isPresent()) {
-        throw new ISE("WTF?! No bucket found for timestamp: %s", key.get());
-      }
-      Interval interval = intervalOptional.get();
+      Interval interval = config.getGranularitySpec().getSegmentGranularity().bucket(new DateTime(key.get()));
```

Comment:
Hmm, the original code looks a bug to me because non-contiguous intervals should be accepted as in IndexTask (https://github.com/druid-io/druid/blob/master/indexing-service/src/main/java/io/druid/indexing/common/task/IndexTask.java#L438-L442). This usually happens when reindexing old data for specified intervals.